test/processor/Package.test.ts

```diff
@@ -110,5 +112,98 @@ describe('Package', () => {
+      slicingPath.caretPath = 'slicing.discriminator[0].path';
+      slicingPath.value = '$this';
+      const slicingOrdered = new ExportableCaretValueRule('value[x]');
+      slicingOrdered.caretPath = 'slicing.ordered';
+      slicingOrdered.value = false;
+      const slicingRules = new ExportableCaretValueRule('value[x]');
+      slicingRules.caretPath = 'slicing.rules';
+      slicingRules.value = new fshtypes.FshCode('open');
+
+      profile.rules.push(slicingType, slicingPath, slicingOrdered, slicingRules);
+      const myPackage = new Package();
+      myPackage.add(profile);
+      myPackage.optimize(processor);
+      expect(profile.rules).toHaveLength(4);
+    });
+
+    it('should not remove caret value rules that define slicing on a non-choice element', () => {
+      const profile = new ExportableProfile('SlicedProfile');
+      profile.parent = 'Observation';
+      const slicingType = new ExportableCaretValueRule('note');
+      slicingType.caretPath = 'slicing.discriminator[0].type';
+      slicingType.value = new fshtypes.FshCode('type');
+      const slicingPath = new ExportableCaretValueRule('note');
+      slicingPath.caretPath = 'slicing.discriminator[0].path';
+      slicingPath.value = '$this';
+      const slicingOrdered = new ExportableCaretValueRule('note');
+      slicingOrdered.caretPath = 'slicing.ordered';
+      slicingOrdered.value = false;
+      const slicingRules = new ExportableCaretValueRule('note');
+      slicingRules.caretPath = 'slicing.rules';
+      slicingRules.value = new fshtypes.FshCode('open');
+      profile.rules.push(slicingType, slicingPath, slicingOrdered, slicingRules);
+      const myPackage = new Package();
+      myPackage.add(profile);
+      myPackage.optimize(processor);
+      expect(profile.rules).toHaveLength(4);
+    });
+
+    it('should not remove caret value rules on a choice element that apply nonstandard choice slicing', () => {
+      const profile = new ExportableProfile('SlicedProfile');
+      profile.parent = 'Observation';
+      const slicingType = new ExportableCaretValueRule('value[x]');
+      slicingType.caretPath = 'slicing.discriminator[0].type';
+      slicingType.value = new fshtypes.FshCode('value');
+      const slicingPath = new ExportableCaretValueRule('value[x]');
+      slicingPath.caretPath = 'slicing.discriminator[0].path';
+      slicingPath.value = 'Observation.code';
+      const slicingOrdered = new ExportableCaretValueRule('value[x]');
+      slicingOrdered.caretPath = 'slicing.ordered';
+      slicingOrdered.value = true;
+      const slicingRules = new ExportableCaretValueRule('value[x]');
+      slicingRules.caretPath = 'slicing.rules';
+      slicingRules.value = new fshtypes.FshCode('closed');
+      profile.rules.push(slicingType, slicingPath, slicingOrdered, slicingRules);
+      const myPackage = new Package();
+      myPackage.add(profile);
+      myPackage.optimize(processor);
+      expect(profile.rules).toHaveLength(4);
+    });
```

Comment:
If you agree w/ my suggestion above that rule removal should be all or nothing, we should add a test that differs from the standard choice slicing by only one rule.